### PR TITLE
Docs: Caching application state/ hydrated bloc

### PR DESCRIPTION
**Pull request by valentinallavayol**
## Description
Added documentation about `hydrated_bloc` and how its used to cache application state (feed, articles, etc) across app restarts and crashes.

## Type of Change

- [ ] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [x] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: docs/docs/flutter_development/cache_application_state.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+---
+sidebar_position: 10
+description: Learn how the toolkit caches your news applications's state.
+---
+
+# Cache your application state
+
+[Hydrated Bloc](https://pub.dev/packages/hydrated_bloc) is a extension to the BLoC pattern. It helps automatically persist and restore BLoC states, ensuring that the app's state is retained across app restarts or crashes.
+
+The project relies on `hydrated_bloc` to persist the state of the following BLoCs:
```
Comment (scarletteliza) [inline docs/docs/flutter_development/cache_application_state.md:10]:
```suggestion
[Hydrated Bloc](https://pub.dev/packages/hydrated_bloc) is a extension to the BLoC pattern. It helps automatically persist and restore bloc states, ensuring that the app's state is retained across app restarts or crashes.

The project relies on `hydrated_bloc` to persist the state of the following blocs:
```

File: docs/docs/flutter_development/cache_application_state.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+- `feed_bloc`: Persists the feed state. It contains the list of feed articles fetched from the API.
+- `article_bloc` : Persists each article information fetched from the API.
+- `categories_bloc`: Persists all feed categories fetched from the API.
+- `theme_mode_bloc` : Persists the theme mode of the app selected by the user.
+
+## How it works
+
+`hydrated_bloc` package uses their own `hydrated_storage` to persist the state of the BLoCs on the application's side. It's enabled per default.
+
+Upon launching the application, if the feed state is empty, it retrieves the feed articles from the API. In contrast, if the feed state is not empty, it displays the cached feed articles. As the user scrolls to the end of the feed page, older feed articles are fetched from the API and added to the feed state that will be persisted.
```
Comment (scarletteliza) [inline docs/docs/flutter_development/cache_application_state.md:21]:
```suggestion
The `hydrated_bloc` package uses its own `hydrated_storage` to persist the state of blocs on the application side. It's enabled on default.

Upon launching the application, if the feed state is empty, it retrieves the feed articles from the API. If the feed state is not empty, it displays the cached feed articles. As the user scrolls to the end of the feed page, older feed articles are fetched from the API and added to the feed state that will be persisted.
```

File: docs/docs/flutter_development/cache_application_state.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+Actions such as opening the app from background or terminated state will not affect the state of the BLoCs. Those will be persisted and restored when the user restarts the app.
+
+In order for the user to see the most recent articles, they must refresh the feed by pulling down the feed page.
+
+If there any errors while fetching the feed articles, the user will be notified by a 'Network Error' screen. The user can retry fetching the articles by tapping on the 'Retry' button.
```
Comment (scarletteliza) [inline docs/docs/flutter_development/cache_application_state.md:27]:
```suggestion
Actions such as opening the app from background or terminated state will not affect the state of the blocs. Those will be persisted and restored when the user restarts the app.

In order for the user to see the most recent articles, they must refresh the feed by pulling down the feed page.

If there any errors while fetching the feed articles, the user will be notified by a 'Network Error' screen. The user can retry fetching the articles by tapping on the 'Retry' button.
```

File: docs/docs/flutter_development/cache_application_state.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+## Debug mode caching
+
+On the project, `hydrated_bloc`'s caching is automatically disabled for debug mode. Every restart of the application will clear the `hydrated_bloc`'s storage state, so no state will be restored. In order to enable it, the following code must be removed from the `bootstrap.dart` file:
+
```
Comment (scarletteliza) [inline docs/docs/flutter_development/cache_application_state.md:32]:
```suggestion
In this project, `hydrated_bloc` caching is automatically disabled for debug mode. Every restart of the application will clear the `hydrated_bloc` storage state, so no state will be restored. In order to enable it, the following code must be removed from the `bootstrap.dart` file:

```

File: docs/docs/flutter_development/cache_application_state.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
+---
+sidebar_position: 10
+description: Learn how the toolkit caches your news applications's state.
```
Comment (scarletteliza) [inline docs/docs/flutter_development/cache_application_state.md:3]:
```suggestion
description: Learn how the toolkit caches your news application's state.
```

File: docs/docs/flutter_development/cache_application_state.md [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+---
+
+# Cache your application state
+
+[Hydrated Bloc](https://pub.dev/packages/hydrated_bloc) is a extension to the BLoC pattern. It helps automatically persist and restore BLoC states, ensuring that the app's state is retained across app restarts or crashes.
```
Comment (marwfair) [inline docs/docs/flutter_development/cache_application_state.md:8]:
```suggestion
[Hydrated Bloc](https://pub.dev/packages/hydrated_bloc) is an extension to the BLoC pattern. It helps automatically persist and restore BLoC states, ensuring that the app's state is retained across app restarts or crashes.
```

File: docs/docs/flutter_development/cache_application_state.md (hunk 3 above)
Comment (marwfair) [inline docs/docs/flutter_development/cache_application_state.md:27]:
```suggestion
If there are any errors while fetching the feed articles, the user will be notified by a 'Network Error' screen. The user can retry fetching the articles by tapping on the 'Retry' button.
```

File: docs/docs/flutter_development/cache_application_state.md [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+The project relies on `hydrated_bloc` to persist the state of the following BLoCs:
+
+- `feed_bloc`: Persists the feed state. It contains the list of feed articles fetched from the API.
+- `article_bloc` : Persists each article information fetched from the API.
+- `categories_bloc`: Persists all feed categories fetched from the API.
+- `theme_mode_bloc` : Persists the theme mode of the app selected by the user.
+
+## How it works
+
+`hydrated_bloc` package uses their own `hydrated_storage` to persist the state of the BLoCs on the application's side. It's enabled per default.
```
Comment (marwfair) [inline docs/docs/flutter_development/cache_application_state.md:19]:
```suggestion
`hydrated_bloc` package uses its own `hydrated_storage` to persist the state of the BLoCs on the application's side. It is enabled by default.
```

**Review by marwfair — COMMENTED** [review]
This is looking great! Just left a few grammar related comments.

**Review by scarletteliza — APPROVED** [review]

**Pull request merged**